### DISCUSSION TITLE
Rust Socket.io client

### DISCUSSION THREAD

**Discussion by BrianWiz** [Totodore/socketioxide#405] [General]
Hey there,

I'm curious if there's any goto clients written in Rust? I'm currently doing some research for implementing my own social features for video games like friends lists, parties, etc. I think socket.io really makes that interesting because of it's Javascript ecosystem. For example it would be really awesome to have a phone app connected to the same server that the game itself connects to. However I'm looking at supporting the Bevy engine (a Rust game engine) primarily.

Probably the most promising looking one is this: https://github.com/1c3t3a/rust-socketio

But I'm very curious if folks are using something else and what that may be, or if the socketioxide team has any interest in doing a Rust client as well.

**Comment by Totodore** [comment]
Hey,

Currently there is only `rust-socketio`, however it lacks some features. I _might_ write a client implementation that would share some common code with socketioxide. Hence the split with multiple crates. But currently it is clearly not the priority as there is a lot of improvement areas for socketioxide (as a socket.io server implementation).

**Reply by BrianWiz** [reply]
Thanks for your quick response!

Well, I guess I'll barrel ahead and see how this pans out. 

While I'm here, how far off are you folks from getting a Redis adapter working? That's going to seriously rock!

**Reply by Totodore** [reply]
Ideally in the first quarter of 2025. But I'm not certain, it depends on my free time and the issues I might encounter.
